### ISSUE TITLE
[Feature Request] Opt-In Telemetry

### ISSUE THREAD

**Issue by 0x6b-dev** [FOSSBilling/FOSSBilling#38]
We will need a mechanism to know what modules are actually being used and worth putting development time into.

Would also be interesting to know how many active installs are there, and what versions they are using (including PHP and MySQL).

Stick -> Carrot = If people don't opt in, their module may not be supported.

**Comment by John-S4** [comment]
Makes sense, but it needs to be explicitly opt-in. Both for ethical reasons and to avoid violating GDPR (and similar).

**Comment by jaapmarcus** [comment]
You might in "privacy" issues. 

Usage can be monitored with the download if we split up everything in different repo's

**Comment by John-S4** [comment]
> Usage can be monitored with the downloads/installs if we split up everything in different repo's

That would only track downloads rather than active installs though?

**Comment by 0x6b-dev** [comment]
I would like to see:

Active installs (how many people are actually using it)
Module usage (what modules and payment gateways people are using)

Fully anonymous of course, totally opt in and totally transparent.

Not interested in having anybody's personal data or violating any laws.

Check this out:
https://github.com/metabase/metabase

**Comment by John-S4** [comment]
I'm not familiar with Metabase but it looks interesting. 

I don't have a problem with any telemetry implementation as long as we ensure that it is opt-in because anything else is just a highway to a world of issues.

**Comment by evrifaessa** [comment]
We also should publish and visualize the data we collect like Steam does in its hardware survey.

https://store.steampowered.com/hwsurvey/Steam-Hardware-Software-Survey-Welcome-to-Steam
